fpga_interchange/examples/chipdb.cmake

```diff
@@ -0,0 +1,112 @@
+function(create_rapidwright_device_db)
+    set(options)
+    set(oneValueArgs part)
+    set(multiValueArgs)
+
+    cmake_parse_arguments(
+        create_rapidwright_device_db
+        "${options}"
+        "${oneValueArgs}"
+        "${multiValueArgs}"
+        ${ARGN}
+    )
+
+    set(part ${create_rapidwright_device_db_part})
+    set(rapidwright_device_db ${CMAKE_CURRENT_BINARY_DIR}/${part}.device)
+    add_custom_command(
+        OUTPUT ${rapidwright_device_db}
+        COMMAND
+            RAPIDWRIGHT_PATH=${RAPIDWRIGHT_PATH}
+            ${RAPIDWRIGHT_PATH}/scripts/invoke_rapidwright.sh
+            com.xilinx.rapidwright.interchange.DeviceResourcesExample
+            ${part}
+        DEPENDS
+            ${RAPIDWRIGHT_PATH}/scripts/invoke_rapidwright.sh
+    )
+
+    add_custom_target(rapidwright-${part}-device DEPENDS ${rapidwright_device_db})
+    set_property(TARGET rapidwright-${part}-device PROPERTY LOCATION ${rapidwright_device_db})
+endfunction()
+
+function(generate_chipdb)
+    set(options)
+    set(oneValueArgs part)
+    set(multiValueArgs)
+
+    cmake_parse_arguments(
+        generate_chipdb
+        "${options}"
+        "${oneValueArgs}"
+        "${multiValueArgs}"
+        ${ARGN}
+    )
+
+    set(part ${generate_chipdb_part})
+
+    create_rapidwright_device_db(part ${part})
+
+    set(rapidwright_device_target rapidwright-${part}-device)
+    get_property(rapidwright_device_loc TARGET ${rapidwright_device_target} PROPERTY LOCATION)
+
+    set(constraints_device ${CMAKE_CURRENT_BINARY_DIR}/${part}_constraints.device)
+    add_custom_command(
+        OUTPUT ${constraints_device}
+        COMMAND
+            python3 -mfpga_interchange.patch
+                --schema_dir ${INTERCHANGE_SCHEMA_PATH}
+                --schema device
+                --patch_path constraints
+                --patch_format yaml
+                ${rapidwright_device_loc}
+                ${PYTHON_INTERCHANGE_PATH}/test_data/series7_constraints.yaml
+                ${constraints_device}
+        DEPENDS
+            ${rapidwright_device_target}
+    )
+
+    add_custom_target(constraints-${part}-device DEPENDS ${constraints_device})
+
+    set(constraints_luts_device ${CMAKE_CURRENT_BINARY_DIR}/${part}_constraints_luts.device)
+    add_custom_command(
+        OUTPUT ${constraints_luts_device}
+        COMMAND
+            python3 -mfpga_interchange.patch
```

Comment:
other arches tend to use `${PYTHON_EXECUTABLE}` rather than `python3` here to allow it to be overridden in some cases if needed

Reply:
Sure, done.